src/main/java/com/beerjournal/breweriana/collection/persistence/UserCollectionRepository.java

```diff
@@ -120,16 +111,33 @@ int updateItem(Item item) {
         return writeResult.getN();
     }
 
-    private Page<Item> findAllNotIn(Set<String> userItemsNames, PageRequest pageRequest, String name, String category) {
-        Criteria criteria = Criteria.where("name").not().in(userItemsNames);
-        if (!Strings.isNullOrEmpty(name))
-            criteria = criteria.andOperator(Criteria.where("name").regex(Pattern.compile("^" + name)));
-        if (!Strings.isNullOrEmpty(category))
-            criteria = criteria.andOperator(Criteria.where("type").regex(Pattern.compile("^" + category)));
+    private Page<Item> findAllByParams(Criteria criteria,
+                                       PageRequest pageRequest,
+                                       Map<String, String> allRequestParams,
+                                       String sortBy,
+                                       String sortType) {
+        List<Criteria> filterCriterias = new ArrayList<>();
+
+        for (Map.Entry<String, String> entry : allRequestParams.entrySet()) {
```

Comment:
forEach isn't possible to use here, maybe you know better approach? As for me for is enough.